### PR TITLE
[tusk-mastodon] Adding tusk-mastodon types

### PR DESCRIPTION
**Pull request by fishcharlie**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tusk-mastodon/index.d.ts
1. types/tusk-mastodon/tusk-mastodon-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bc9e299c92b50de2402feda5adfa034442deb3e2
</p>

File: types/tusk-mastodon/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    formEncodeParams(params: GeneralObject): string;
+}
+
+interface TuskMastodonOptions {
+    "access_token": string;
+    /**
+     * The base URL of the Mastodon instance to connect to. Defaults to `https://mastodon.social/api/v1/`.
+     */
+    "api_url": string;
```
Comment (peterblazejewicz) [inline types/tusk-mastodon/index.d.ts:23]: this one looks like optional:
https://github.com/skullzarmy/Tusk/blob/master/lib/mastodon.js#L19
```suggestion
    "api_url?": string;
```

File: types/tusk-mastodon/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+declare class TuskMastodonInstance {
+    constructor(options: TuskMastodonOptions);
+
+    put(path: string, params: GeneralObject): Promise<GeneralObject>;
```
Comment (peterblazejewicz) [inline types/tusk-mastodon/index.d.ts:4]: all these `params` seem to be optional parameter:
https://github.com/skullzarmy/Tusk/blob/master/lib/mastodon.js#L19 (etc)

File: types/tusk-mastodon/index.d.ts (hunk 1 above)
Comment (fishcharlie) [reply types/tusk-mastodon/index.d.ts]: Fixed in 3c1885d5cfeaf2f22b1e560da62793bf14d3cea4

File: types/tusk-mastodon/index.d.ts (hunk 2 above)
Comment (fishcharlie) [reply types/tusk-mastodon/index.d.ts]: Fixed in b514bfa3bbd9b586228961070f40026c674c1396

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@fishcharlie thanks!

**Comment by fishcharlie** [comment]
Ready to merge

**Pull request merged**